doc/architecture/architecture.rst

```diff
@@ -1,16 +1,190 @@
+===================================
+
+One way to descibe Ostro OS architecture is to build a stack from
+hardware platform up to an IoT Application layer. This stack is
+illustrated in Fig 1.
+
+.. image::images/ostro_os_architecture.png
+
+In the following, each layer is explained in more detail.
+
+IoT Applications
+----------------
+The IoT applications layer consist of all applications that use the
+underlying platform. Ostro OS itself currently does not
+implement any concrete IoT use-cases via applications but many sample
+applications are provided to demonstrate the OS services.
+
+Programming Interfaces
+-----------------------
+Ostro OS provides various application runtimes to allow the end-user
+more choices to use in their IoT applications.
+
+* `Soletta Project`_ is an open source
+framework for making IoT applications. The project provides libraries
+to make it quick and easy to write software for IoT devices. The
+applications can we written using a flow-based programing (FBP) or
+more traditionally as C applications linking to Soletta's C based
+platform APIs.
+
+* `Node.JS`_ is a popular JavaScript runtime to
+run (IoT) web applications. Ostro OS provides the Node.JS runtime
+and selected JavaScript APIs (as Node.JS modules) to build IoT
+applications. The set of APIs include, e.g., a Javascript API for
+the OIC specification. The Javascript API set is kept aligned with
+the API set provided by Soletta.
+
+* In addition to Soletta and Node.JS, common programming languages/
+runtimes in IoT space are supported. Ostro OS provides Python runtime
+and Java Platform (OpenJDK8) that are popular in the IoT space.
+
+Frameworks
+----------
+
+Frameworks provide an abstraction to platform services. For making
+application deployment easy, the Ostro Application
+Framework :ref:`application_framework` provides tooling for
+application writers to get the applications running in an Ostro OS
+based image.
+
+The Ostro Application framework implements a systemd generator that
+parses the application manifest files to systemd .service.
+
+`IoTivity`_ TODO
+
+Services
+--------
+
+The responsibility of system services is to bring up the system,
+manage wireless connections, and process' IPC. The components
+used are commonly used in all open source OSes: systemd, connman,
+bluez, and D-Bus.
+
+In addition to systemd and connectivity, Ostro OS comes with
+:ref: `software-update` technology that helps to support software
+updates to deployed devices.
+
+Base Libraries
+--------------
+
+Standard Linux base libraries are used in Ostro...
+
+
+Linux Kernel and Hardware Adaptation
+-------------------------------------
+
+The hardware BSPs for Ostro OS run the Linux Kernel. The kernel
+provides the necessary drivers and hardware adaptation.
+
+Sensors and connectivity are critical for IoT devices. Therefore,
+Ostro OS maintains a dedicated page: :ref: `hardware` that describes
+how various peripherals can be run with :ref:`quick_start/platforms`
```

Comment:
Out of curiosity (question to @dbkinder ), do we need to have the "quick_start/" prefix here or would :ref:`platforms` work equally well?